gp_predict.f95

```diff
@@ -777,6 +783,14 @@ subroutine gpSparse_initialise(this, from, error)
       a(1:n_globalY,:) = transpose(c_subYY_sqrtInverseLambda)
       a(n_globalY+1:,:) = factor_c_subYsubY
 
+      if (condition_number_norm_opt(1:1) /= ' ') then
+         allocate(acopy(size(a, 1),size(a, 2)))
+         acopy = a
+         call matrix_condition_number(acopy, rcond, condition_number_norm_opt(1:1))
```

Comment:
Maybe we should have picked it up at the implementation of the condition number calculator:
- it may be nicer to use a function, as it only returns one value
- if calling it has a side effect (i.e. changing acopy) I think the allocation of the temporary array should happen there, not on the calling side. I'd expect intent(in) arguments here.

Reply:
The reasoning for allocating the copy outside the procedure was that the user can decide for themselves if they want to make a copy or live with the changed matrix. Feels more general for libAtoms. On the other hand it's only used by GAP.

Reply:
I have not implemented the ScaLAPACK version of the condition number calculation yet. The condition number of a submatrix can differ a lot from the full one so I would just print a warning that it is not implemented for distributed matrices.